### PR TITLE
refactor: add tr_variant_serde

### PR DESCRIPTION
**Pull request by ckerr**
add a serializer / deserializer helper class for `tr_variant`.

### CODE REVIEW

File: tests/libtransmission/json-test.cc [hunk 1]
```diff
@@ -173,15 +174,14 @@ TEST_P(JSONTest, test1)
...
         "        }\n"
         "    }\n"
-        "}\n"
-    };
+        "}\n"sv;
 
-    tr_variant top;
-    EXPECT_TRUE(tr_variantFromBuf(&top, TR_VARIANT_PARSE_JSON | TR_VARIANT_PARSE_INPLACE, in));
+    auto serde = tr_variant_serde::json();
+    auto top = serde.parse(Input).value_or(tr_variant{});
```
Comment (tearfur) [inline tests/libtransmission/json-test.cc:180]: *(minor)* Missing `.inplace()`

File: tests/libtransmission/makemeta-test.cc [hunk 2]
```diff
@@ -234,19 +234,18 @@ TEST_F(MakemetaTest, announceSingleTracker)
 
     // generate the torrent and parse it as a variant
     EXPECT_EQ(nullptr, builder.make_checksums().get());
-    auto top = tr_variant{};
-    auto const benc = builder.benc();
-    EXPECT_TRUE(tr_variantFromBuf(&top, TR_VARIANT_PARSE_BENC | TR_VARIANT_PARSE_INPLACE, benc));
+    auto top = tr_variant_serde::benc().parse(builder.benc());
```
Comment (tearfur) [inline tests/libtransmission/makemeta-test.cc:237]: ditto

File: tests/libtransmission/makemeta-test.cc [hunk 3]
```diff
@@ -265,22 +264,21 @@ TEST_F(MakemetaTest, announceMultiTracker)
...
```
Comment (tearfur) [inline tests/libtransmission/makemeta-test.cc:267]: ditto

File: libtransmission/rpc-server.cc [hunk 4]
```diff
@@ -361,18 +361,13 @@ void rpc_response_func(tr_session* /*session*/, tr_variant* content, void* user_
 
 void handle_rpc_from_json(struct evhttp_request* req, tr_rpc_server* server, std::string_view json)
 {
-    auto top = tr_variant{};
-    auto const have_content = tr_variantFromBuf(&top, TR_VARIANT_PARSE_JSON | TR_VARIANT_PARSE_INPLACE, json);
+    auto top = tr_variant_serde::json().inplace().parse(json);
```
[the remark this answers is not in the document]
Comment (ckerr) [reply libtransmission/rpc-server.cc]: 110f405c4d8f0b075fc765097e199ab6e5f5c4ba

File: tests/libtransmission/json-test.cc (hunk 1 above)
Comment (ckerr) [reply tests/libtransmission/json-test.cc]: good catch. 3fd6ef3934169741261e977d7a7186ae240c909a

File: tests/libtransmission/makemeta-test.cc (hunk 2 above)
Comment (ckerr) [reply tests/libtransmission/makemeta-test.cc:237]: I don't think that's correct --the string returned by `builder.benc()` goes out-of-scope after the parsing is done

File: tests/libtransmission/makemeta-test.cc (hunk 3 above)
Comment (ckerr) [reply tests/libtransmission/makemeta-test.cc:267]: same `builder.benc()` issue

**Review by tearfur — APPROVED** [review]

**Pull request merged**